### DISCUSSION TITLE
To know about click event on image in custom formatter

### DISCUSSION THREAD

**Discussion by vysakhtk886** [ghiscoding/Angular-Slickgrid#863] [Q&A] [answered]
I 'am trying to call a click event on image tag in custom formatter,  But it didn't work as expected
Is it possible to add click event on a image?.
![slickgrid](https://user-images.githubusercontent.com/81738337/144841117-4f2bd1b3-e674-4e8a-8e7e-8da606e23b84.png)

**Comment by ghiscoding** [comment, marked as answer]
because that's not the way to do it, you should always avoid adding any JS code in a Formatter because that is an open door to scripting attack (XSS). Funny enough you're not the only one thinking this way recently and it all came up recently because I closed that door in a recent SlickGrid [PR #652](https://github.com/6pac/SlickGrid/pull/652) that will sanitize (remove) any JS code from a Formatter and other areas of SlickGrid and a few people got surprised as well. 

The correct way to do it in SlickGrid is by using any of the SlickGrid [events](https://github.com/6pac/SlickGrid/blob/master/slick.grid.js#L5976) that it exposes publicly, but an even easier way in Angular-Slickgrid is to use the column definition property [onCellClicked](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/interfaces/column.interface.ts#L213) that I added to the Column interface (it's a simple wrapper of the SlickGrid `onClick` event but it's specific to whichever column you add it to). I'm probably missing documentation for that (you could help on that), but that's typically the way I do it.

```ts
this.columnDefinitions = [{ 
  id: 'firstName', /*....*/ 
  onCellClick: (event, args) => {
     const row, cell, dataView, columnDef, dataContext = args;
  }
}];
```
I added 2 common event wrappers in the Column interface, `onCellClicked` and `onCellChanged`
